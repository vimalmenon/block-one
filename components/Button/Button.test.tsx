import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import {Button} from "./";

describe("Button Component", () => {
    test("Button: Snapshot testing Primary Type", () => {
        const wrapper = shallow(<Button type="primary" />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    test("Button: Snapshot testing Secondary Type", () => {
        const wrapper = shallow(<Button type="secondary" />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    test("Button: Snapshot testing disabled", () => {
        const wrapper = shallow(<Button disabled={true} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    test("Button: Snapshot testing disabled", () => {
        const onClick= jest.fn();
        const wrapper = shallow(<Button onClick={onClick} />);
        wrapper.find("button").simulate("click");
        expect(onClick).toHaveBeenCalled();
    });
});