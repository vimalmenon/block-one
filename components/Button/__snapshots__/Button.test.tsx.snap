// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Button: Snapshot testing Primary Type 1`] = `
<button
  className="px-4 py-2 focus:outline-none rounded-md bg-blue-500 text-white"
  disabled={false}
  onClick={[Function]}
/>
`;

exports[`Button Component Button: Snapshot testing Secondary Type 1`] = `
<button
  className="px-4 py-2 focus:outline-none rounded-md bg-yellow-300 text-black"
  disabled={false}
  onClick={[Function]}
/>
`;

exports[`Button Component Button: Snapshot testing disabled 1`] = `
<button
  className="px-4 py-2 focus:outline-none rounded-md bg-gray-600 text-gray-200 "
  disabled={true}
  onClick={[Function]}
/>
`;
